File: 3467.transform-array-by-parity.java
/*
 * @lc app=leetcode id=3467 lang=java
 *
 * [3467] Transform Array by Parity
 */

// @lc code=start
class Solution {
    public int[] transformArray(int[] nums) {
        int even = 0;
        int odd = 0;
        for (int i : nums) {
            if (i % 2 == 0) {
                even += 1;
            }
            else {
                odd += 1;
            }
        }
        int[] result = new int[odd + even];
        for (int i = even; i < even + odd; i++) {
            result[i] = 1;
        }
        return result;
    }
}
// @lc code=end

File: 3467.transform-array-by-parity.py
#
# @lc app=leetcode id=3467 lang=python
#
# [3467] Transform Array by Parity
#

# @lc code=start
class Solution(object):
    def transformArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        even = odd = 0
        for i in nums:
            if i % 2:
                odd += 1
            else:
               even += 1
        result = [0] * (even)
        for i in range(odd):
            result.append(1)
        return result        
        
# @lc code=end

